lang/cs/Org.Apache.REEF.Wake/Remote/Impl/RemoteConnectionRetryHandler.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using Microsoft.Practices.TransientFaultHandling;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Wake.Remote.Parameters;
+
+namespace Org.Apache.REEF.Wake.Remote.Impl
+{
+    /// <summary>
+    /// Handles the retry logic to connect to remote endpoint
+    /// </summary>
+    internal class RemoteConnectionRetryHandler : IConnectionRetryHandler
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(RemoteConnectionRetryHandler));
+        private readonly int _retryCount;
+        private readonly TimeSpan _timeIntervalInMs;
+        private RetryPolicy<AllErrorsTransientStrategy> _retryPolicy;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="retryCount">Number of times to retry connection</param>
+        /// <param name="timeIntervalInMs">Time interval between retries in milli seconds</param>
+        [Inject]
+        internal RemoteConnectionRetryHandler([Parameter(typeof(ConnectionRetryCount))] int retryCount,
+            [Parameter(typeof(SleepTimeInMs))] int timeIntervalInMs)
+        {
+            _retryCount = retryCount;
+            _timeIntervalInMs = TimeSpan.FromMilliseconds(timeIntervalInMs);
+            _retryPolicy = new RetryPolicy<AllErrorsTransientStrategy>(
+                    new FixedInterval("ConnectionRetries", _retryCount, _timeIntervalInMs, true));
+            _retryPolicy.Retrying += (sender, args) =>
```

Comment:
RemoteManager, StreamingRemoteManager and Transport tests all use it so one try case is sort of verified with those tests....For retry logic I am not sure what test we can write since on server side there is no way to reject connection if you are listening. I though of sleep based tests where server is started a bit after client so that client has failed sometimes, but I am not sure these sleep based tests are good ideas.

In general, I tested with IMRU.Examples on YARN cluster and verified that retry logic kicks in and works.

@markusweimer  any comment on this?